File: arduinobot_ws/src/arduinobot_py_examples/arduinobot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimplePublisher(Node):

    def __init__(self):
        super().__init__("simple_publisher")
        self.pub_ = self.create_publisher(String, "chatter", 10)
        self.counter_ = 0
        self.get_logger().info("Node Publisher Ready")

    def execute(self, rate):
        self.get_logger().info("Publishing at %dHz" % rate)

        rate = self.create_rate(rate)
        while rclpy.ok():
            msg = String()
            msg.data = "Hello ROS 2 - counter: %d" % self.counter_
            self.pub_.publish(msg)
            self.counter_ += 1
            rate.sleep()


def main(args=None):
    rclpy.init(args=args)

    simple_publisher = SimplePublisher()
    simple_publisher.execute(1)

    rclpy.spin(simple_publisher)

    simple_publisher.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: arduinobot_ws/src/arduinobot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>


class SimplePublisher : public rclcpp::Node
{
public:
  SimplePublisher() : Node("simple_publisher"), counter_(0)
  {
    pub_ = create_publisher<std_msgs::msg::String>("chatter", 10);
    RCLCPP_INFO(get_logger(), "Node Publisher Ready");
  }

  void execute(int rate)
  {
    RCLCPP_INFO_STREAM(get_logger(), "Publishing at " << rate << "Hz");
    rclcpp::Rate loop_rate(rate);
    while (rclcpp::ok())
    {
      auto message = std_msgs::msg::String();
      message.data = "Hello ROS 2 - counter:" + std::to_string(counter_++);
      pub_->publish(message);
      loop_rate.sleep();
    }
  }

private:
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
  unsigned int counter_;
};


int main(int argc, char* argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<SimplePublisher>();
  node->execute(1);
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}